File: python-sdk/ade_sidechain/bridge.py
"""Bridge client for cross-chain operations"""

from typing import Dict
from enum import Enum
from dataclasses import dataclass
from .client import AdeSidechainClient


class BridgeStatus(Enum):
    """Bridge operation status"""
    PENDING = 'pending'
    LOCKED = 'locked'
    RELAYED = 'relayed'
    COMPLETED = 'completed'
    FAILED = 'failed'


@dataclass
class DepositInfo:
    """Information about a bridge deposit"""
    deposit_id: str
    from_chain: str
    to_chain: str
    amount: int
    token: str
    sender: str
    recipient: str
    status: BridgeStatus
    timestamp: int


@dataclass
class WithdrawalInfo:
    """Information about a bridge withdrawal"""
    withdrawal_id: str
    from_chain: str
    to_chain: str
    amount: int
    token: str
    sender: str
    recipient: str
    status: BridgeStatus
    timestamp: int


class BridgeClient:
    """Client for bridge operations"""

    def __init__(self, client: AdeSidechainClient):
        self.client = client

    def deposit(
        self,
        from_chain: str,
        amount: int,
        token_address: str
    ) -> Dict[str, str]:
        """Initiate a deposit from another chain"""
        return self.client.bridge_deposit(
            from_chain=from_chain,
            amount=amount,
            token_address=token_address,
        )

    def withdraw(
        self,
        to_chain: str,
        amount: int,
        recipient: str
    ) -> Dict[str, str]:
        """Initiate a withdrawal to another chain"""
        return self.client.bridge_withdraw(
            to_chain=to_chain,
            amount=amount,
            recipient=recipient,
        )

File: cli/src/commands/bridge.rs
use anyhow::Result;
use colored::*;
use serde_json::json;
use crate::commands::rpc::RpcClient;

pub async fn handle_bridge_command(action: crate::BridgeAction, rpc_url: &str) -> Result<()> {
    let client = RpcClient::new(rpc_url);

    match action {
        crate::BridgeAction::Deposit { from_chain, amount, token } => {
            println!("Initiating bridge deposit...");
            
            let result = client.call("bridgeDeposit", Some(json!({
                "fromChain": from_chain,
                "amount": amount,
                "tokenAddress": token
            }))).await?;
            
            println!("{}", "Deposit initiated".green());
            println!("{}: {}", "Deposit ID".bold(), result.get("depositId").unwrap());
            println!("{}: {}", "Signature".bold(), result.get("signature").unwrap());
        }
        crate::BridgeAction::Withdraw { to_chain, amount, recipient } => {
            println!("Initiating bridge withdrawal...");
            
            let result = client.call("bridgeWithdraw", Some(json!({
                "toChain": to_chain,
                "amount": amount,
                "recipient": recipient
            }))).await?;
            
            println!("{}", "Withdrawal initiated".green());
            println!("{}: {}", "Withdrawal ID".bold(), result.get("withdrawalId").unwrap());
            println!("{}: {}", "Signature".bold(), result.get("signature").unwrap());
        }
        crate::BridgeAction::Status { id } => {
            let result = client.call("getBridgeStatus", Some(json!({ "id": id }))).await?;
            
            println!("{}", "Bridge Operation Status".bold().green());
            println!("{}", serde_json::to_string_pretty(&result)?);
        }
    }

    Ok(())
}

